pipeline/relation_embedder/relation_embedder/src/main/scala/weco/pipeline/relation_embedder/BatchProcessor.scala

```diff
@@ -103,39 +102,38 @@ class BatchProcessor(
 object BatchProcessor {
 
   def apply(
-    config: Config
+    config: RelationEmbedderConfig
   )(
     implicit actorSystem: ActorSystem,
     ec: ExecutionContext,
     materializer: Materializer
   ): BatchProcessor = {
 
     val identifiedIndex =
-      Index(config.requireString("es.merged-works.index"))
+      Index(config.mergedWorkIndex)
 
-    val esClient = ElasticBuilder.buildElasticClient(config)
+    val esClient = ElasticBuilder.buildElasticClient(config.elasticConfig)
```

Comment:
Nice!  I think this is the line that really proves the value of the approach.

Previously, we were just handing all the config over to buildElasticClient, and letting it pick what it wants.  Now we can be much more targeted about it.